custom_components/sleep_as_android/__init__.py

```diff
@@ -288,6 +304,12 @@ async def subscribe_2021_07(
 
         if self._ha_version is None:
             await self._get_version()
+        if self._ha_version >= AwesomeVersion("2024.12.0"):
+            self._subscription_state = await subscribe_2024_12(
+                self.hass,
+                self._subscription_state,
+                topic,
+            )
         if self._ha_version >= AwesomeVersion("2022.3.0"):
```

Comment:
FYI I was still receiving an error after the update related to the 2022 version. Changing this to an elif resolved the issue